### PR TITLE
refactor(basic-auth): achieving 100% coverage and improving performance

### PR DESCRIPTION
**Pull request by moliveiraz**
Initially, I intended to reach **100% coverage** and improve **code readability.**

### CODE REVIEW

File: src/middleware/basic-auth/index.ts [hunk 1]
```diff
@@ -6,65 +6,71 @@ import { decodeBase64 } from '../../utils/encode'
 
 const CREDENTIALS_REGEXP = /^ *(?:[Bb][Aa][Ss][Ii][Cc]) +([A-Za-z0-9._~+/-]+=*) *$/
 const USER_PASS_REGEXP = /^([^:]*):(.*)$/
-const utf8Decoder = new TextDecoder()
+
+interface AuthOptions {
```
Comment (yusukebe) [inline src/middleware/basic-auth/index.ts:10]: `BasicAuthOptions` would be better, although it's a bit verbose.

File: src/middleware/basic-auth/index.ts [hunk 2]
```diff
@@ -6,65 +6,71 @@ import { decodeBase64 } from '../../utils/encode'
...
   }
-
-  return { username: userPass[1], password: userPass[2] }
 }
 
 export const basicAuth = (
-  options: { username: string; password: string; realm?: string; hashFunction?: Function },
-  ...users: { username: string; password: string }[]
+  options: AuthOptions,
+  ...users: ShortAuthOptions[]
```
Comment (yusukebe) [inline src/middleware/basic-auth/index.ts:36]: Also, I think `ShortBasicAuthOptions` is better.

**Comment by yusukebe** [comment]
Hi @moliveiraz !

Awesome work! I've commented some issues but, almost looks good! One thing which I'd like to do. Please do `yarn denoify` to generate file for Deno. And add&commit&push it. Thanks!

**Comment by moliveiraz** [comment]
Hi, @yusukebe!

My pull request didn't pass the tests, why?

**Comment by yusukebe** [comment]
Hi @moliveiraz 

You have deleted some files for Deno that weren't necessary to delete. Could you reopen this PR and run `denoify` again?

**Pull request closed without merging**